File: merg.js
function mergeSort(arr) {
    if (arr.length <= 1) {
        return arr;
    }

    const mid = Math.floor(arr.length / 2);
    const leftHalf = arr.slice(0, mid);
    const rightHalf = arr.slice(mid);

    const leftSorted = mergeSort(leftHalf);
    const rightSorted = mergeSort(rightHalf);

    return merge(leftSorted, rightSorted);
}

function merge(left, right) {
    const result = [];
    let leftIdx = 0, rightIdx = 0;

    while (leftIdx < left.length && rightIdx < right.length) {
        if (left[leftIdx] < right[rightIdx]) {
            result.push(left[leftIdx]);
            leftIdx++;
        } else {
            result.push(right[rightIdx]);
            rightIdx++;
        }
    }

    while (leftIdx < left.length) {
        result.push(left[leftIdx]);
        leftIdx++;
    }

    while (rightIdx < right.length) {
        result.push(right[rightIdx]);
        rightIdx++;
    }

    return result;
}

// Example 
const arr = [10, 7, 8, 9, 1, 5];
const sortedArr = mergeSort(arr);
console.log("Sorted array:", sortedArr);

File: merg.py
def merge_sort(arr):
    if len(arr) <= 1:
        return arr

    mid = len(arr) // 2
    left_half = arr[:mid]
    right_half = arr[mid:]

    left_half = merge_sort(left_half)
    right_half = merge_sort(right_half)

    return merge(left_half, right_half)

def merge(left, right):
    result = []
    left_idx, right_idx = 0, 0

    while left_idx < len(left) and right_idx < len(right):
        if left[left_idx] < right[right_idx]:
            result.append(left[left_idx])
            left_idx += 1
        else:
            result.append(right[right_idx])
            right_idx += 1

    while left_idx < len(left):
        result.append(left[left_idx])
        left_idx += 1

    while right_idx < len(right):
        result.append(right[right_idx])
        right_idx += 1

    return result

# Example 
if __name__ == "__main__":
    arr = [10, 7, 8, 9, 1, 5]
    sorted_arr = merge_sort(arr)
    print("Sorted array:", sorted_arr)
